src/shared/ui/picker.ts

```diff
@@ -145,3 +149,220 @@ export function verifySinglePickerOutput<T extends vscode.QuickPickItem>(choices
+            detail: localize('AWS.picker.dynamic.noItemsFound.detail', 'Click here to go back'),
+            alwaysShow: true,
+        }
+        this.errorItem = {
+            label: localize('AWS.picker.dynamic.errorNode.label', 'There was an error retrieving more items.'),
+            alwaysShow: true,
+        }
+
+        const quickPickButtons = this.pickerOptions.buttons || []
+        quickPickButtons.push(this.pauseButton, this.playButton)
+        if (this.pickerOptions.isRefreshable) {
+            quickPickButtons.push(this.refreshButton)
+        }
+        // is this the correct impl? If so, is this the correct icon?
+        // e.g. this or quickpick item at the end of the list?
+        if (this.pickerOptions.paginationType) {
+            quickPickButtons.push(this.paginationButton)
+        }
+
+        // TODO: Create default buttons for load next page, refresh
+        // TODO: Set a global throttling flag that will optionally display said load next page button
+        this.quickPick = createQuickPick<vscode.QuickPickItem>({
+            options: {
+                ...this.pickerOptions.options,
+                onDidSelectItem: item => {
+                    // pause any existing execution
+                    this.isPaused = true
+
+                    // pass existing onDidSelectItem through if it exists and isn't a base case
+                    if (this.pickerOptions.options?.onDidSelectItem) {
+                        this.pickerOptions.options.onDidSelectItem(item)
+                    }
+                },
+            },
+            items: this.items,
+            buttons: quickPickButtons,
+        })
+
+        this.moreItemsRequest.event(() => this.loadItems())
+    }
+
+    /**
+     * Prompts the user with the quick pick specified by the constructor.
+     * Always attempts to load new results from the iteratingAwsCall, even if the call has been exhausted.
+     * If the call picker was previously paused, unpauses it.
+     */
+    public async promptUser(): Promise<vscode.QuickPickItem[] | undefined> {
+        if (!this.isDone) {
+            this.moreItemsRequest.fire()
+        }
+        return await promptUser<vscode.QuickPickItem>({
+            picker: this.quickPick,
+            onDidTriggerButton: (button, resolve, reject) => {
+                // pause any existing execution
+                this.isPaused = true
+                if (button === this.refreshButton) {
+                    this.isPaused = true
+                    this.refreshQuickPick()
+                    this.moreItemsRequest.fire()
+                    return
+                }
+                if (button === this.paginationButton) {
+                    this.isPaused = false
+                    this.moreItemsRequest.fire()
+                    return
+                }
+                // TODO: remove debug buttons
+                if (button === this.pauseButton) {
+                    this.isPaused = true
```

Comment:
could remove the `isPaused = true` assignments since that is done on line 280. Or if the intention is to be explicit, then the assignment should also be done in the fall-through case (line 302)

Reply:
This was for testing purposes. I don't think we'll have a reason to keep this around.